File: backend/app/models/source_table.py
from datetime import datetime
from typing import Optional
from sqlalchemy import BigInteger, String, SmallInteger, DateTime, Text, Integer
from sqlalchemy.orm import Mapped, mapped_column
from app.database import Base


class SourceTable(Base):
    __tablename__ = "source_tables"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(128), nullable=False)
    description: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    scope: Mapped[int] = mapped_column(SmallInteger, default=2)  # 1=public 2=private
    owner_id: Mapped[Optional[int]] = mapped_column(BigInteger, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)


class TableColumn(Base):
    __tablename__ = "table_columns"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    table_id: Mapped[int] = mapped_column(BigInteger, nullable=False)
    column_name: Mapped[str] = mapped_column(String(128), nullable=False)
    data_type: Mapped[str] = mapped_column(String(64), nullable=False)
    comment: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    sort_order: Mapped[int] = mapped_column(Integer, default=0)

File: backend-java/src/main/java/com/ddh/agent/domain/model/table/SourceTable.java
package com.ddh.agent.domain.model.table;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("source_tables")
public class SourceTable {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String description;
    /** 1=public 2=private */
    private Integer scope;
    private Long ownerId;
    private LocalDateTime createdAt;
}
